metricbeat/module/windows/perfmon/pdh_windows.go

```diff
@@ -82,28 +74,24 @@ func (q *Handle) ReadData(firstFetch bool) (common.MapStr, PdhError) {
 	result := common.MapStr{}
 
 	for _, v := range q.counters {
-		groupVal := make(map[string]interface{})
-		for _, v1 := range v.Counters {
-			err := _PdhGetFormattedCounterValue(v1.counter, PdhFmtDouble, q.counterType, &v1.displayValue)
-			if err != ERROR_SUCCESS {
-				switch err {
-				case PDH_CALC_NEGATIVE_DENOMINATOR:
-					{
-						//Sometimes counters return negative values. We can ignore this error. See here for a good explanation https://www.netiq.com/support/kb/doc.php?id=7010545
-						groupVal[v1.counterName] = 0
-						continue
-					}
-				default:
-					{
-						return nil, PdhError(err)
-					}
+		err := _PdhGetFormattedCounterValue(v.counter, PdhFmtDouble, q.counterType, &v.displayValue)
+		if err != ERROR_SUCCESS {
+			switch err {
+			case PDH_CALC_NEGATIVE_DENOMINATOR:
+				{
+					//Sometimes counters return negative values. We can ignore this error. See here for a good explanation https://www.netiq.com/support/kb/doc.php?id=7010545
+					result[v.counterName] = 0
+					continue
+				}
+			default:
+				{
+					return nil, PdhError(err)
 				}
 			}
-			doubleValue := (*float64)(unsafe.Pointer(&v1.displayValue.LongValue))
-			groupVal[v1.counterName] = *doubleValue
-
 		}
-		result[v.GroupName] = groupVal
+		doubleValue := (*float64)(unsafe.Pointer(&v.displayValue.LongValue))
+
+		result[v.counterName] = *doubleValue
```

Comment:
You should use here `result.Put(v.counterName, *doubleValue)`, see https://github.com/elastic/beats/blob/master/libbeat/common/mapstr.go#L105 This will create sub documents for each dot which will make it compatible with elasticsearch 2.x.